File: SubscriptionApi/SubscriptionApi/Commands/CommandFactory.cs
using System.Collections.Generic;
using SubscriptionApi.Authorization;
using SubscriptionApi.Dto;
using SubscriptionApi.ResponseBuilders;
using SubscriptionApi.Subscriptions;

namespace SubscriptionApi.Commands
{
  public interface ICommandFactory
  {
    Command CreateFrom(NewSubscriptionParametersDto parameters, SubscriptionStartResponseBuilder responseBuilder);
    Command CreateFrom(StoppedSubscriptionParametersDto parameters, SubscriptionStopResponseBuilder responseBuilder);
  }

  public class CommandFactory : ICommandFactory
  {
    private readonly SubscriptionsModifyOperations _subscriptions;
    private readonly AuthorizationStructure _authorizationStructure;
    private readonly ISubscriptionFactory _subscriptionFactory;
    private readonly IAssetQueriesFactory _assetQueriesFactory;
    private readonly Log _log;
    private readonly SubscriptionDataCorrectnessCriteria _dataCorrectnessCriteria;

    public CommandFactory(
      SubscriptionsModifyOperations subscriptions, 
      AuthorizationStructure authorizationStructure, 
      ISubscriptionFactory subscriptionFactory, 
      SubscriptionDataCorrectnessCriteria dataCorrectnessCriteria, 
      IAssetQueriesFactory assetQueriesFactory, 
      Log log)
    {
      _subscriptions = subscriptions;
      _authorizationStructure = authorizationStructure;
      _subscriptionFactory = subscriptionFactory;
      _dataCorrectnessCriteria = dataCorrectnessCriteria;
      _assetQueriesFactory = assetQueriesFactory;
      _log = log;
    }

    public Command CreateFrom(NewSubscriptionParametersDto parameters, SubscriptionStartResponseBuilder responseBuilder)
    {
      var assetQueries = _assetQueriesFactory.CreateFrom(parameters.Requests);
      return 
        new ExceptionLoggedCommand(_log,
          new AdapterFromSubscriptionCommandToCommand(
            new SubscriptionStartCommandFromApi(
              parameters, 
              _authorizationStructure, 
              responseBuilder, 
              _subscriptionFactory, 
              _subscriptions,
              assetQueries)));
    }

    public Command CreateFrom(StoppedSubscriptionParametersDto parameters, SubscriptionStopResponseBuilder responseBuilder)
    {
      return 
        new ExceptionLoggedCommand(_log,
          new AdapterFromSubscriptionCommandToCommand(
            new SubscriptionStopCommandFromApi(
              parameters, 
              responseBuilder, 
              _subscriptions, 
              _dataCorrectnessCriteria, 
              _authorizationStructure)));
    }
  }
}

File: Java/factories/_01_no_factories/src/main/java/Commands/CommandFactory.java
package commands;

import authorization.AuthorizationStructure;
import dto.NewSubscriptionParametersDto;
import dto.StoppedSubscriptionParametersDto;
import queries.AssetQuery;
import responseBuilders.SubscriptionStartResponseBuilder;
import responseBuilders.SubscriptionStopResponseBuilder;
import subscriptions.DataCorrectnessCriteria;
import subscriptions.ISubscriptionFactory;
import subscriptions.SubscriptionsModifyOperations;
import other.Log;

import java.util.List;

import static java.util.Arrays.asList;

public class CommandFactory implements ICommandFactory {
    private final SubscriptionsModifyOperations subscriptions;
    private final AuthorizationStructure authorizationStructure;
    private final ISubscriptionFactory subscriptionFactory;
    private final IAssetQueriesFactory assetQueriesFactory;
    private final Log log;
    private final DataCorrectnessCriteria dataCorrectnessCriteria;

    public CommandFactory(
        SubscriptionsModifyOperations subscriptions,
        AuthorizationStructure authorizationStructure,
        ISubscriptionFactory subscriptionFactory,
        DataCorrectnessCriteria dataCorrectnessCriteria,
        IAssetQueriesFactory assetQueriesFactory,
        Log log) {
        this.subscriptions = subscriptions;
        this.authorizationStructure = authorizationStructure;
        this.subscriptionFactory = subscriptionFactory;
        this.dataCorrectnessCriteria = dataCorrectnessCriteria;
        this.assetQueriesFactory = assetQueriesFactory;
        this.log = log;
    }

    public Command createFrom(NewSubscriptionParametersDto parameters, SubscriptionStartResponseBuilder responseBuilder) {
        List<AssetQuery> assetQueries = assetQueriesFactory
            .createFrom(asList(parameters.requests));
        return
            new ExceptionLoggedCommand(log,
                new AdapterFromSubscriptionCommandToCommand(
                    new SubscriptionStartCommandFromApi(
                        parameters,
                        authorizationStructure,
                        responseBuilder,
                        subscriptionFactory,
                        subscriptions,
                        assetQueries)));
    }

    public Command createFrom(StoppedSubscriptionParametersDto parameters, SubscriptionStopResponseBuilder responseBuilder) {
        return
            new ExceptionLoggedCommand(log,
                new AdapterFromSubscriptionCommandToCommand(
                    new SubscriptionStopCommandFromApi(
                        parameters,
                        responseBuilder,
                        subscriptions,
                        dataCorrectnessCriteria,
                        authorizationStructure)));
    }
}
